clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -151,6 +153,45 @@ public HubConnectionState getConnectionState() {
         return connectionState;
     }
 
+
+    public CompletableFuture startAsync() throws Exception {
+        if (connectionState != HubConnectionState.DISCONNECTED) {
```

Comment:
We're going to need a lock here (we probably needed it before too). If I call `connection.start()` twice without waiting for the first one, the second one should block until the first one finishes. See [HttpConnection.StartAsyncCore](https://github.com/aspnet/SignalR/blob/adc5706bcb2b3cb85e98e303999eb9a778426f7c/src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs#L195)

Reply:
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/Lock.html ?